hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/UsersManager.java

```diff
@@ -72,7 +72,7 @@
 
   // To detect whether there is a change in user count for every user-limit
   // calculation.
-  private AtomicLong latestVersionOfUsersState = new AtomicLong(0);
+  private long latestVersionOfUsersState = 0;
```

Comment:
The only place where latestVersionOfUsersState was being called without a lock was in isRecomputeNeeded. Please also see the comment in isRecomputeNeeded().